x/millions/types/message_deposit_edit.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ sdk.Msg = &MsgDepositEdit{}
+
+func NewMsgDepositEdit(depositorAddr string, poolID uint64, depositID uint64) *MsgDepositEdit {
+	return &MsgDepositEdit{
+		PoolId:           poolID,
+		DepositId:        depositID,
+		DepositorAddress: depositorAddr,
+	}
+}
+
+func (msg MsgDepositEdit) Route() string {
+	return RouterKey
+}
+
+func (msg MsgDepositEdit) Type() string {
+	return "DepositEdit"
+}
+
+func (msg *MsgDepositEdit) GetSigners() []sdk.AccAddress {
+	depositor := sdk.MustAccAddressFromBech32(msg.GetDepositorAddress())
+	return []sdk.AccAddress{depositor}
+}
+
+func (msg *MsgDepositEdit) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgDepositEdit) ValidateBasic() error {
```

Comment:
I think we can safely add this static check (from the `Deposit.ValidateBasic` method):

```

	if msg.GetWinnerAddress() != "" {
		if _, err := sdk.AccAddressFromBech32(msg.GetWinnerAddress()); err != nil {
			return ErrInvalidWinnerAddress
		}
	}

	if msg.WinnerAddress != "" && msg.WinnerAddress != msg.DepositorAddress && msg.IsSponsor {
		return ErrInvalidSponsorWinnerCombo
	}

```